pkg/controllers/management/k3sbasedupgrade/k3sbased_upgrade_handler.go

```diff
@@ -160,13 +177,29 @@ func (h *handler) deployK3sBasedUpgradeController(clusterName string, isK3s, isR
 				ProjectName:     appProjectName,
 				TargetNamespace: systemUpgradeNS,
 			},
+			Status: v33.AppStatus{
+				// we should be using helm 3 if we can
+				HelmVersion: common.HelmV3,
+			},
+		}
+
+		// if we're creating the app for the first time, and we are using a version
+		// above or equal to 1.25 we need to explicitly disable PSPs
+		if is125OrAbove {
+			desiredApp.Spec.ValuesYaml = fmt.Sprintf(ConditionalPSPValues, "false")
+		} else {
+			desiredApp.Spec.ValuesYaml = fmt.Sprintf(ConditionalPSPValues, "true")
 		}
 
 		// k3s upgrader doesn't exist yet, so it will need to be created
 		if _, err = appClient.Create(desiredApp); err != nil {
 			return err
 		}
 	} else {
+		// we should enforce the use of helm 3 from now on
```

Comment:
Setting Helm 3 on the app status is fine, I just see conflicting messages between this and line 181. Do we want to use helm 3 if possible or always regardless of k8s version?